Mailnag/backends/imap.py

```diff
@@ -93,15 +93,15 @@ def list_messages(self):
 				logging.debug('Folder %s in status %s | Data: %s', (folder, status, data))
 				continue # Bugfix LP-735071
 			for num in data[0].split():
-				typ, msg_data = conn.fetch(num, '(BODY.PEEK[HEADER])') # header only (without setting READ flag)
+				typ, msg_data = conn.uid('FETCH', num, '(BODY.PEEK[HEADER])') # header only (without setting READ flag)
 				for response_part in msg_data:
 					if isinstance(response_part, tuple):
 						try:
 							msg = email.message_from_bytes(response_part[1])
 						except:
 							logging.debug("Couldn't get IMAP message.")
 							continue
-						yield (folder, msg)
+						yield (folder, msg, num.decode("utf-8"))
```

Comment:
I see, that is of course an issue. Maybe it would be a way to add some kind of dummy number to the other backends? I don't see any other way of letting mailnag know about this imap specific id though